taca/storage/storage.py

```diff
@@ -64,6 +81,93 @@ def archive_to_swestore(days, run=None):
+    """
+        Remove project/run that have been closed more than 'days'
+        from the given 'site' on uppmax
+
+        :param str site: site where the cleanup should be performed
+        :param int days: number of days to check for closed projects
+    """
+    config = get_config()
+    LOG = get_logger()
+    root_dir = config.get('cleanup').get(site)
+    deleted_log = config.get('cleanup').get('deleted_log')
+    assert os.path.exists(os.path.join(root_dir,deleted_log)), "Log directory {} doesn't exist in {}".format(deleted_log,root_dir)
+    log_file = os.path.join(root_dir,"{fl}/{fl}.log".format(fl=deleted_log))
+    PRO_RE = '[a-zA-Z]+\.[a-zA-Z]+_\d{2}_\d{2}'
+    db_config = config.get('statusdb',{})
+
+    # make a connection for project db #
+    pcon = statusdb.ProjectSummaryConnection(**db_config)
+    assert pcon, "Could not connect to project database in StatusDB"
+
+    if site != "archive":
+        ## work flow for cleaning up illumina/analysis ##
+        projects = [ p for p in os.listdir(root_dir) if re.match(PRO_RE,p) ]
+        list_to_delete = get_closed_projects(projects,pcon,days,LOG)
+    else:
+        ##work flow for cleaning archive ##
+        list_to_delete = []
+        runs = [ r for r in os.listdir(root_dir) if re.match(filesystem.RUN_RE,r) ]
+        with filesystem.chdir(root_dir):
+            for run in runs:
+                fcid = run.split('_')[3]
+                fcid = fcid if re.search('-',fcid) else fcid[1:]
+                ## fetch and prarse samplesheet to get projects ##
+                try:
+                    ssheet = [c for c in os.listdir(run) if re.search('{}.csv|SampleSheet.csv'.format(fcid),c)][0]
+                except IndexError:
+                    LOG.warn("Could not find expected samplesheet for run {}, so SKIPPING..")
+                    continue
+                with open(os.path.join(run,ssheet),'r') as ss:
+                    try:
+                        run_projs = list(set([ d['SampleProject'].replace('__','.') for d in csv.DictReader(ss) ]))
+                    except:
+                        LOG.warn("Samplesheet is not in expected format for run {}".format(run))
+                        continue
+                ## check if all projects of the run have been closed ##
+                if len(run_projs) == len(get_closed_projects(run_projs,pcon,days)):
+                    list_to_delete.append(run)
+                else:
+                    LOG.warn("All projects that were ran on {} are not closed, so SKIPPING".format(run))
+                    continue
+
+    ## delete and log
+    for item in list_to_delete:
+        if dry_run:
+            LOG.info('Will remove {} from {}'.format(item,root_dir))
+            continue
+        try:
+            shutil.rmtree(os.path.join(root_dir,item))
+            LOG.info('Removed project {} from {}'.format(item,root_dir))
+            mode = 'a' if os.path.exists(log_file) else 'w'
```

Comment:
I tried that first, but for some reason it dint work :( 
It just kept overwriting the file. But will give it a try again and see more carefully :+1: